feat: add tools of quiz voice upload

Norton-Lin opened the pull request:
新增测试题语言上传工具
close #287

Norton-Lin committed:
feat: add tools of quiz voice upload

新增测试题语言上传工具

Norton-Lin committed:
style: format quiz_voice_upload.py

lin594 commented on tools/README.md:
```diff
@@ -0,0 +1 @@
+#   小工具集合
```
后面写点小介绍呀  比如xxx 工具

lin594 commented on tools/quiz_voice_upload/quiz_voice_upload.py:
```diff
@@ -0,0 +1,44 @@
+import json
```
在这个文件夹外面添加一个 readme 吧  怎么使用的 干啥的 都说一下呗

Norton-Lin replied:
这个不急，等你批斗完代码再写233333

lin594 commented on tools/quiz_voice_upload/quiz_voice_upload.py:
```diff
@@ -0,0 +1,44 @@
+
+from environs import Env
+
+env = Env()
```
这个env是干啥的 我一点都没有理解？

Norton-Lin replied:
读环境变量 账户密码

lin594 replied:
哦 后知后觉 你是要做“数据脱敏”是吧、、
才想起来这个事情
（原本想的就是账号密码的变量等于一个乱码x 23333）

Norton-Lin replied:
乱码他咋登录23333

lin594 commented on tools/quiz_voice_upload/quiz_voice_upload.py:
```diff
@@ -0,0 +1,44 @@
+password = env.str("password")
+data = {"username": username, "password": password}
+token = requests.post(
+    f"https://api.pxm.edialect.top/login", data=json.dumps(data)
```
网址无论正式服务器还是测试服务器  都应该用一个变量 类似于 `base_url` 这样的方式去组合

lin594 commented on tools/quiz_voice_upload/quiz_voice_upload.py:
```diff
@@ -0,0 +1,44 @@
+).json()["token"]
+url = input("请输入文件夹路径:")
+#   url = os.path.pardir
+file = os.listdir(url)
```
既然是文件夹里的东西 那应该是 `files` 吧？

lin594 commented on tools/quiz_voice_upload/quiz_voice_upload.py:
```diff
@@ -0,0 +1,44 @@
+        result["quiz"]["voice_source"] = f"https://cos.edialect.top/quizzes/{id}{str}"
+        requests.put(
+            f"https://api.pxm.edialect.top/quizzes/{id}",
+            data=json.dumps(result),
```
dumps 是啥 挺好奇的

lin594 commented on tools/quiz_voice_upload/quiz_voice_upload.py:
```diff
@@ -0,0 +1,44 @@
+        res = requests.get(
+            f"https://api.pxm.edialect.top/quizzes/{id}", headers={"token": token}
+        ).json()
+        #   测试服
+        #   res = requests.get(f"https://api.pxm.test.edialect.top/quizzes/{id}", headers={'token': token}).json()
+        #   本地
+        #   res = requests.get(f"http://127.0.0.1:8000/quizzes/{id}", headers={'token': token}).json()
```
用了 base url  这些注释都没用了吧x

Norton-Lin committed:
feat: add base_url to quiz_voice_upload

新增 base_url 于 quiz_voice_upload.py
新增 小工具readme介绍

lin594 commented:
话说 能用嘛
代码应该问题不大
你弄个到测试网站上试一试？

lin594 commented:
@Norton-Lin 测试过嘛 测试过我就合并了？

Norton-Lin committed:
feat： media upload

存在问题：
qcloud_cos.cos_exception.CosClientError: EOF occurred in violation of protocol (_ssl.c:997)
测试时，website.view 中的upload_file函数同样存在此问题

lin594 commented:
不对 傻逼了
这个就是直接改数据库 不要上传的

不过 upload文件如果坏掉也要修复一下的  我估计是测试服务器上面的 上传没配置好 正式服务器的可以用

Norton-Lin commented:
23333，那我那个upload还要不要
单纯的修改数据库脚本是没问题的（不过用的时候得改一下默认url，现在用的是测试服）

Norton-Lin committed:
refactor： remove upload_file.py

lin594 commented on tools/quiz_voice_upload/upload_file.py:
这个东西为啥还在呀
以及上传的操作直接调用接口就行吧？
测试服的坏掉应该去修那个）

Norton-Lin replied:
噶了噶了

Norton-Lin committed:
refactor: remove upload_file.py
